src/mmt_tcpip/lib/protocols/http2.c

```diff
@@ -267,3 +267,201 @@ int mmt_check_http2(ipacket_t *ipacket, unsigned proto_index) {
 		return 0;
 }
 
+int restore_http2_packet(uint8_t*data_out,const ipacket_t * packet,int proto_offset){
+ 	int header_length=0;
+ 	int offset_header_length = proto_offset -1;
+   	for (int i = offset_header_length; i < offset_header_length+4; i++) {
+      		  header_length = (header_length << 8) | data_out[i];
+   	 }
+   	 header_length=header_length & 0x00FFFFFF;
+ 	int new_length=0;
+   	 for (int i = offset_header_length; i < offset_header_length+4; i++) {
+      		  new_length = (new_length << 8) | packet->data[i];
+   	 }
+   	 new_length=new_length & 0x00FFFFFF;
+   	// printf("new_length %d, header_length %d\n",new_length,header_length);
+   	// printf("data_out %02X packet_data %02X \n",data_out[proto_offset],packet->data[proto_offset]);
+ 	memcpy((uint8_t*)data_out+proto_offset,packet->data+proto_offset,new_length+9);
```

Comment:
Before doing `memcpy`, you need to check
- whether `data_out+proto_offset` passes over the capaiblity of `data_out`, and, 
- whether `new_length+9` is not bigger than the capability of `data_out+proto_offset`